player/lua/ytdl_hook.lua

```diff
@@ -418,23 +418,23 @@ local function formats_to_edl(json, formats, use_all_formats)
         end
 
         local tracks = {}
-        if track.vcodec and track.vcodec ~= "none" then
+        -- "none" means it is not a video
+        -- nil means it is unknown
+        if (o.force_all_formats or track.vcodec) and track.vcodec ~= "none" then
             tracks[#tracks + 1] = {
                 media_type = "video",
                 codec = map_codec_to_mpv(track.vcodec),
             }
         end
-        -- Tries to follow the strange logic that vcodec unset means it's
-        -- an audio stream, even if acodec is sometimes unset.
-        if (#tracks == 0) or (track.acodec and track.acodec ~= "none") then
+        if (o.force_all_formats or track.acodec) and track.acodec ~= "none" then
             tracks[#tracks + 1] = {
                 media_type = "audio",
                 codec = map_codec_to_mpv(track.acodec) or
                         ext_map[track.ext],
             }
         end
         if #tracks == 0 then
-            return nil
+            goto SKIP_TRACK
```

Comment:
Wasn't `goto` only added in Lua 5.2? This won't work in Lua 5.1 or LuaJit builds.

Reply:
You're right, I didn't know that. Good catch

Reply:
I've just tried goto in LuaJit and it seems to be one of the Lua 5.2 things that are supported in LuaJit.

Reply:
Ah sorry, I have actually used it before in LuaJit but it seems that slipped my mind. But yes, we need to support Lua 5.1, I have needed to modify one of my scripts before because a user was on that version.